wagtail/admin/tests/pages/test_parent_page_chooser_view.py

```diff
@@ -117,3 +119,45 @@ def test_choosing_parent_with_unsupported_subpage_type(self):
             "You cannot create a page of type &quot;Event page&quot; under &quot;%s&quot;."
             % page_with_limited_subtypes.get_admin_display_title(),
         )
+
+    def test_skip_if_only_one_valid_parent(self):
+        self.assertEqual(EventIndex.objects.count(), 1)
+
+        with mock.patch.object(EventPage, "allowed_parent_page_models") as mock_method:
+            mock_method.return_value = [EventIndex]
+            self.assertEqual(EventPage.allowed_parent_page_models(), [EventIndex])
+            response = self.client.get(self.view_url)
+
+        self.assertRedirects(
+            response,
+            reverse(
+                "wagtailadmin_pages:add",
+                args=("tests", "eventpage", EventIndex.objects.first().pk),
+            ),
+        )
+
+    def test_skip_if_user_only_has_permission_for_one_parent(self):
```

Comment:
I added this test per my comment, to ensure the redirect also works when it's caused by limited permissions (and not due to the `parent_page_types` on the model)